test/cmd/newapp.sh

```diff
@@ -193,6 +193,10 @@ os::cmd::expect_failure_and_text "echo 'S P A C E S=test' | oc new-build --binar
 # new-build - check that we can set build-args in DockerStrategy 
 os::cmd::expect_success_and_text "oc new-build ${OS_ROOT}/test/testdata/build-arg-dockerfile --build-arg 'foo=bar' --to 'test' -o jsonpath='{.items[?(@.kind==\"BuildConfig\")].spec.strategy.dockerStrategy.buildArgs[?(@.name==\"foo\")].value}'" 'bar'
 
+# check that we cannot set build-args in a non-DockerStrategy build
+os::cmd::expect_failure "oc new-build https://github.com/openshift/ruby-hello-world --strategy=source --build-arg 'foo=bar'"
+os::cmd::expect_failure "oc new-build https://github.com/openshift/ruby-ex --build-arg 'foo=bar'"
```

Comment:
use expect_failure_and_text for both of these.  make sure you're getting the error you expect.